File: beatonma-django/common/microformats.py
from enum import StrEnum


class HCard(StrEnum):
    # dates
    dt_bday = "dt-bday"
    # values
    p_name = "p-name"
    p_locality = "p-locality"
    p_region = "p-region"
    p_country_name = "p-country-name"
    # links
    u_email = "u-email"
    u_logo = "u-logo"
    u_photo = "u-photo"
    u_url = "u-url"


class HAdr(StrEnum):
    p_locality = "p-locality"
    p_region = "p-region"
    p_country_name = "p-country-name"


type Microformat = HCard | HAdr

File: frontend/src/microformats.ts
export enum H {
  "h-card" = "h-card",
  "h-adr" = "h-adr",
}

/**
 * Known fields for an h-card.
 */
export enum HCard {
  "dt-bday" = "dt-bday",
  "p-name" = "p-name",
  "p-locality" = "p-locality",
  "p-region" = "p-region",
  "p-country-name" = "p-country-name",
  "u-email" = "u-email",
  "u-logo" = "u-logo",
  "u-photo" = "u-photo",
  "u-url" = "u-url",
}

export type Microformat = keyof typeof H | keyof typeof HCard;
